src/MSBuild/XMake.cs

```diff
@@ -1413,9 +1413,18 @@ internal static bool BuildProject(
                 bool isPreprocess = preprocessWriter != null;
                 bool isTargets = targetsWriter != null;
 
+                ILogger[] evaluationLoggers =
+                    [
+                        // all of the loggers that are single-node only
+                        .. loggers,
+                        // all of the central loggers for multi-node systems. These need to be resilient to multiple calls
+                        // to Initialize
+                        .. distributedLoggerRecords.Select(d => d.CentralLogger)
+                    ];
```

Comment:
This is what gives us access to eval-time errors for our console/file/terminal loggers